frame/staking/src/lib.rs

```diff
@@ -732,6 +731,143 @@ impl<AccountId, Balance: Default + HasCompact> Default for Exposure<AccountId, B
+				page_total,
+				others: chunk
+					.iter()
+					.map(|c| IndividualExposure { who: c.who.clone(), value: c.value })
+					.collect(),
+			});
+		}
+
+		(
+			ExposureOverview {
+				total: self.total,
+				own: self.own,
+				nominator_count: self.others.len() as u32,
+				page_count: exposure_pages.len() as PageIndex,
+			},
+			exposure_pages,
+		)
+	}
+}
+
+/// A snapshot of the stake backing a single validator in the system.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct ExposurePage<AccountId, Balance: HasCompact> {
+	/// The total balance of this chunk/page.
+	#[codec(compact)]
+	pub page_total: Balance,
+	/// The portions of nominators stashes that are exposed.
+	/// TODO(ank4n): BoundedVec touches lot of code, skip for now.
+	pub others: Vec<IndividualExposure<AccountId, Balance>>,
+}
+
+impl<AccountId, Balance: Default + HasCompact> Default for ExposurePage<AccountId, Balance> {
+	fn default() -> Self {
+		Self { page_total: Default::default(), others: vec![] }
+	}
+}
+
+/// An overview of stake backing a single validator.
+///
+/// It, in combination with a list of `ExposurePage`s, can be used to reconstruct a full `Exposure`
+/// struct. This is useful for cases where we want to query a single page of `Exposure`s.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct ExposureOverview<Balance: HasCompact> {
+	/// The total balance backing this validator.
+	#[codec(compact)]
+	pub total: Balance,
+	/// The validator's own stash that is exposed.
+	#[codec(compact)]
+	pub own: Balance,
+	/// Number of nominators backing this validator.
+	pub nominator_count: u32,
+	/// Number of pages of backers.
+	pub page_count: PageIndex,
+}
+
+impl<Balance: Default + HasCompact> Default for ExposureOverview<Balance> {
+	fn default() -> Self {
+		Self {
+			total: Default::default(),
+			own: Default::default(),
+			nominator_count: Default::default(),
+			page_count: Default::default(),
+		}
+	}
+}
+
+/// Extended view of Exposure comprising of `ExposureOverview` and a single page of `ExposurePage`.
+///
+/// This is useful where we need to take into account the validator's own stake and total exposure
+/// in consideration in addition to the individual nominators backing them.
+#[derive(Encode, Decode, RuntimeDebug, TypeInfo, PartialEq, Eq)]
+struct ExposureExt<AccountId, Balance: HasCompact> {
+	exposure_overview: ExposureOverview<Balance>,
+	exposure_page: ExposurePage<AccountId, Balance>,
+}
+
+impl<AccountId, Balance: HasCompact + Copy + AtLeast32BitUnsigned> ExposureExt<AccountId, Balance> {
+	/// Create a new instance of `ExposureExt` from legacy clipped exposures.
+	pub fn from_clipped(exposure: Exposure<AccountId, Balance>) -> Self {
+		Self {
+			exposure_overview: ExposureOverview {
+				total: exposure.total,
+				own: exposure.own,
+				nominator_count: exposure.others.len() as u32,
+				page_count: 1,
```

Comment:
Not sure if relevant, but if the len of `exposure` is 0, `page_count` should be 0. Maybe even better would be to return an option of `self`.

Reply:
The older `ErasStakersClipped` returned a `ValueQuery` which returns a default value of it if it does not exist. I tried to keep the old era reward logic exactly same as before since in 84 eras, we will get rid of all this anyways. Let me know wdyt?

Reply:
Sounds fair yes, especially if we'll remove this code soon.